test/marketplace/MarketPlace.sol

```diff
@@ -50,214 +54,246 @@ contract MarketPlace {
+    // TODO remove the require here? why?
+    require(VaultTracker(mkt.vaultTracker).matureVault(exchangeRate), 'mature vault failed');
 
-    emit Mature(u, m, currentExchangeRate, block.timestamp);
+    emit Mature(p, u, m, exchangeRate, block.timestamp);
 
     return true;
   }
 
   /// @notice Allows Swivel caller to deposit their underlying, in the process splitting it - minting both zcTokens and vault notional.
+  /// @param p Protocol Enum value associated with this market
   /// @param u Underlying token address associated with the market
   /// @param m Maturity timestamp of the market
   /// @param t Address of the depositing user
   /// @param a Amount of notional being added
-  function mintZcTokenAddingNotional(address u, uint256 m, address t, uint256 a) external authorized(swivel) unpaused() returns (bool) {
-    Market memory mkt = markets[u][m];
+  function mintZcTokenAddingNotional(uint8 p, address u, uint256 m, address t, uint256 a) external authorized(swivel) unpaused() returns (bool) {
+    Market memory mkt = markets[p][u][m];
     require(ZcToken(mkt.zcToken).mint(t, a), 'mint zcToken failed');
     require(VaultTracker(mkt.vaultTracker).addNotional(t, a), 'add notional failed');
     
     return true;
   }
 
   /// @notice Allows Swivel caller to deposit/burn both zcTokens + vault notional. This process is "combining" the two and redeeming underlying.
+  /// @param p Protocol Enum value associated with this market
   /// @param u Underlying token address associated with the market
   /// @param m Maturity timestamp of the market
   /// @param t Address of the combining/redeeming user
   /// @param a Amount of zcTokens being burned
-  function burnZcTokenRemovingNotional(address u, uint256 m, address t, uint256 a) external authorized(swivel) unpaused() returns(bool) {
-    Market memory mkt = markets[u][m];
+  function burnZcTokenRemovingNotional(uint8 p, address u, uint256 m, address t, uint256 a) external authorized(swivel) unpaused() returns(bool) {
+    Market memory mkt = markets[p][u][m];
     require(ZcToken(mkt.zcToken).burn(t, a), 'burn failed');
     require(VaultTracker(mkt.vaultTracker).removeNotional(t, a), 'remove notional failed');
     
     return true;
   }
 
   /// @notice Allows (via swivel) zcToken holders to redeem their tokens for underlying tokens after maturity has been reached.
+  /// @param p Protocol Enum value associated with this market
   /// @param u Underlying token address associated with the market
   /// @param m Maturity timestamp of the market
   /// @param t Address of the redeeming user
   /// @param a Amount of zcTokens being redeemed
-  function redeemZcToken(address u, uint256 m, address t, uint256 a) external authorized(swivel) unpaused() returns (uint256) {
-    Market memory mkt = markets[u][m];
+  function redeemZcToken(uint8 p, address u, uint256 m, address t, uint256 a) external authorized(swivel) unpaused() returns (uint256) {
+    Market memory mkt = markets[p][u][m];
```

Comment:
This would be a fix before this PR, but I think we should go with the full word `market` here. Seems like that abbreviation slipped in.